spec/CloudCodeSessionTrigger.spec.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+const Parse = require('parse/node');
+
+describe('CloudCode _Session Trigger tests', () => {
+  it('beforeSave should be okay', async () => {
+    Parse.Cloud.beforeSave('_Session', async req => {
+      const sessionObject = req.object;
+      expect(sessionObject).toBeDefined();
+      expect(sessionObject.get('sessionToken')).toBeDefined();
+      expect(sessionObject.get('createdWith')).toBeDefined();
+      expect(sessionObject.get('user')).toBeDefined();
+      expect(sessionObject.get('user')).toEqual(jasmine.any(Parse.User));
+    });
+    try {
+      // signup a user (internally creates a session)
+      const user = new Parse.User();
+      user.setUsername('some-user-name');
+      user.setPassword('password');
+      await user.signUp();
+    } catch (error) {
+      throw error;
+    }
+  });
+  it('beforeSave should discard any changes', async () => {
```

Comment:
Can we put a blank line between each spec?

Also, can we use the appropriate sentencing pattern:

It(‘should discard any changes in beforeSave’)

If you need to group becaus it’s all about before save, please enclose in a `describe(‘beforeSave’...`